flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -116,14 +118,18 @@
 	private final int getShardIteratorMaxAttempts;
 
 	/**
-	 * Create a new KinesisProxy based on the supplied configuration properties.
+	 * Create a new KinesisProxy based on the supplied configuration and client.
 	 *
 	 * @param configProps configuration properties containing AWS credential and AWS region info
+	 * @param kinesisClient the underlying kinesis client
 	 */
-	private KinesisProxy(Properties configProps) {
+
+	private KinesisProxy(Properties configProps, AmazonKinesisClient kinesisClient) {
+
 		checkNotNull(configProps);
+		checkNotNull(kinesisClient);
 
-		this.kinesisClient = AWSUtil.createKinesisClient(configProps);
```

Comment:
AWS related config properties still need to be considered?

Reply:
I wasn't wild about the proposed change. The config props are needed by this class in addition to configuring the client. In this case, the assumption was that the client was created/configured correctly already.

Reply:
It appears that `ClientConfiguration` has setters for all the properties? So can we come up with a generic prefix based convention for config properties that will be transferred into the client configuration? That would keep the changes contained in `AWSUtil`.

Reply:
Yeah, I was hoping to avoid this, but I don't think we can.